cou/steps/plan.py

```diff
@@ -534,7 +534,9 @@ def _generate_control_plane_plan(
     )
 
     logger.debug("Generation of the control plane upgrade plan complete")
-    control_plane_upgrade_plan = [principal_upgrade_plan, subordinate_upgrade_plan]
+    # Note: relate to issue #466
+    # The subordinate application should be upgrade first before principle
```

Comment:
Something like this may make it clearer:

```suggestion
    # Subordinate charms should be upgraded before principal charms
    # to avoid the subordinates being on an unsupported platform after principal charms are upgraded.
    # See https://github.com/canonical/charmed-openstack-upgrader/issues/466#issuecomment-2209991680 for more information.
```